src/wrapper/jnienv.rs

```diff
@@ -1078,6 +1081,12 @@ impl<'a> JNIEnv<'a> {
         Ok(vec)
     }
 
+    // NB: Oracle doc states that the family of New<PrimitiveType>Array routines does not throw any
+    // exception but just return NULL in case of error. In fact, if one calls any of this functions
+    // with negative value as an argument for the length of array it ends up with the pending
+    // java.lang.NegativeArraySizeException exception. That's why we use macro that checks for
+    // exceptions and NULL.
+    // ??? Wouldn't it be more optimal to check for negative argument before calling managed code???
```

Comment:
The spec also says they [throw](https://docs.oracle.com/javase/10/docs/specs/jni/functions.html#newprimitivetypearray-routines)

> OutOfMemoryError: if the system runs out of memory.

But in that case you will both get `NULL` and a pending exception.

Reply:
It's funny, but [spec for version 8](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#New_PrimitiveType_Array_routines) says nothing about throw :-). Thanks, removed the comment.